instrumentation/apache-elasticjob-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apacheelasticjob/v3_0/ElasticJobTest.java

```diff
@@ -0,0 +1,362 @@
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.testing.assertj.AttributeAssertion;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import org.apache.shardingsphere.elasticjob.api.JobConfiguration;
+import org.apache.shardingsphere.elasticjob.http.props.HttpJobProperties;
+import org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.ScheduleJobBootstrap;
+import org.apache.shardingsphere.elasticjob.reg.base.CoordinatorRegistryCenter;
+import org.apache.shardingsphere.elasticjob.reg.zookeeper.ZookeeperConfiguration;
+import org.apache.shardingsphere.elasticjob.reg.zookeeper.ZookeeperRegistryCenter;
+import org.apache.shardingsphere.elasticjob.script.props.ScriptJobProperties;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class ElasticJobTest {
+
+  private static final int EMBED_ZOOKEEPER_PORT = 4181;
+  private static final String ZOOKEEPER_CONNECTION_STRING = "localhost:" + EMBED_ZOOKEEPER_PORT;
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static CoordinatorRegistryCenter regCenter;
+  private static int port;
+  private static HttpServer httpServer;
+
+  @BeforeAll
+  static void init() throws IOException {
+    EmbedZookeeperServer.start(EMBED_ZOOKEEPER_PORT);
+    regCenter = setUpRegistryCenter();
+    port = PortUtils.findOpenPort();
+    httpServer = HttpServer.create(new InetSocketAddress(port), 0);
+    httpServer.createContext(
+        "/hello",
+        new HttpHandler() {
+          @Override
+          public void handle(HttpExchange exchange) throws IOException {
+            byte[] response = "{\"success\": true}".getBytes(UTF_8);
+            exchange.sendResponseHeaders(HttpURLConnection.HTTP_OK, response.length);
+            exchange.getResponseBody().write(response);
+            exchange.close();
+          }
+        });
+    new Thread(() -> httpServer.start()).start();
+  }
+
+  @AfterAll
+  static void stop() {
+    if (httpServer != null) {
+      httpServer.stop(0);
+    }
+    if (regCenter != null) {
+      regCenter.close();
+    }
+    EmbedZookeeperServer.stop();
+  }
+
+  @Test
+  void testHttpJob() {
+    ScheduleJobBootstrap bootstrap = setUpHttpJob(regCenter);
+    try {
```

Comment:
I'd use `AutoCleanupExtension` to run `bootstrap.shutdown()` so I wouldn't need to wrap the whole method in try catch